seed/countries.js

```diff
@@ -1,7 +1,23 @@
-import { countries } from './data/countries.js';
+import { countries } from './data/countriesAndCitiesAndUniversities.js';
 import Country from '../models/Country.js'
+import City from '../models/City.js'
+import University from '../models/University.js'
+import mongoose from 'mongoose';
 
 export default async () => {
-    // transaction
-    await Country.insertMany(countries)
+    let countryId;
+    const session = await mongoose.startSession()
+    await session.withTransaction(async () => {
+        for await (const country of countries) {
+            countryId = mongoose.Types.ObjectId()
+            new Country({ _id: countryId, name: country.name }).save(function (err, res) {
+            }, { session })
```

Comment:
if there is an error => the callback function is triggered, but countries and cities will continue to work which is wrong behavior.

you have 2 solutions:

1. convert to unmanaged transaction, and if there is an error, you abort the transaction.
2. convert to `async/await` syntax instead of callback function